internal/provider/resource_schema.go

```diff
@@ -834,11 +834,13 @@ func loadSchema(ctx context.Context, d *schema.ResourceData, c *SchemaRegistryRe
 func readSchemaRegistryConfigAndSetAttributes(ctx context.Context, d *schema.ResourceData, c *SchemaRegistryRestClient, subjectName string, schemaIdentifier string) (*sr.Schema, error) {
 	isLatestSchemaBool := isLatestSchema(schemaIdentifier)
 	srSchema, exists, err := loadSchema(ctx, d, c, subjectName, schemaIdentifier)
-	if err != nil {
+	_, resp, err := c.apiClient.SchemasV1Api.GetSchemas(c.apiContext(ctx)).SubjectPrefix(subjectName).Execute()
+	isResource404 := ResponseHasExpectedStatusCode(resp, http.StatusNotFound)
+	if !isResource404 && err != nil {
 		return nil, fmt.Errorf("error reading Schema %q: %s", d.Id(), createDescriptiveError(err))
 	}
 	if !exists {
-		if !d.IsNewResource() {
+		if isResource404 && !d.IsNewResource() {
```

Comment:
Nice!

I'm considering whether we could remove line 837, as loadSchema() performs all API calls, and update loadSchema() instead.